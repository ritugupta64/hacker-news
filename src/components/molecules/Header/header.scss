@import '../../../styles/variable.scss';

.header {
  background-color: $orangeColor;
  display: flex;
  font-size: 14px;
  color: $whiteColor;
  align-items: center;
  padding: .5rem;
  &__comments, &__vote-count {
    flex: 2;
  }
  &__up-vote {
    flex: 1;
    margin-right: 5px;
  }
  &__news-details {
    flex: 10;
  }
}
