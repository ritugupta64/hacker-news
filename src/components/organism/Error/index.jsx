import React from 'react'
import {Heading} from '../../atoms/Heading/heading'

export const Error = () => {
    return <Heading headingtype="h1">Page not found</Heading>
}