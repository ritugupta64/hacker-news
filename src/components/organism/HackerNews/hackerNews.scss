@import '../../../styles/variable.scss';

.btn-container {
  display: flex;
  margin: 10px 10px;
  justify-content: flex-end;
  &__btn {
    color: $orangeColor;
    background: none;
    border: none;
    display: inline-block;
    font-size: 16px;
    line-height: 20px;
    padding: 5px;
    &:first-child:after{
      content: '|';
      padding-left: 10px;
    }
    &.disabled{
      opacity: .5;
    }
  }
}
