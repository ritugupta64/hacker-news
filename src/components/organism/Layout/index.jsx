import React from "react";
import PropTypes from "prop-types";

import './layout.scss'

export const Layout = ({ children }) => {
  return <main>{children}</main>;
};

Layout.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.element,
    PropTypes.elementType,
    PropTypes.array,
    PropTypes.object,
  ]).isRequired,
};
