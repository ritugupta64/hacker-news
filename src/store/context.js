import React from 'react'
import PropTypes from 'prop-types'

export const GlobalContext = React.createContext()

export const GlobalContextProvider = ({ data, children }) => {
  return (
    <GlobalContext.Provider value={data}>{children}</GlobalContext.Provider>
  )
}

GlobalContextProvider.propTypes = {
    children: PropTypes.oneOfType([
        PropTypes.element,
        PropTypes.elementType,
        PropTypes.array,
        PropTypes.object,
      ]).isRequired,
      data: PropTypes.shape().isRequired
}