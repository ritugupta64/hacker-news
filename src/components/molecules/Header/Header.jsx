import React from 'react'
import { Link } from 'react-router-dom'
import logo from '../../../y18.png'

import './header.scss'

export const Header = () => {
  return (
    <header className="header">
      <Link to="/">
        <img src={logo} width="18" alt="Hacker News" className="header__img" />
      </Link>
      <h1>
        {' '}
        <Link to="/">Hacker News</Link>
      </h1>
    </header>
  )
}
