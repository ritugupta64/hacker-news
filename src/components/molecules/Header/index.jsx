import React from 'react'

import './header.scss'

export const Header = ({ comments, voteCount, upVote, newsDetails }) => {
  return (
    <header className="header">
      <div className="header__comments">{comments}</div>
      <div className="header__vote-count">{voteCount}</div>
      <div className="header__up-vote">{upVote}</div>
      <div className="header__news-details">{newsDetails}</div>
    </header>
  )
}
