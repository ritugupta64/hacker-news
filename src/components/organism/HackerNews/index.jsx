import React, { useState, useEffect, useContext } from 'react'

import { HACKER_NEWS } from '../../../API'
import { HackerNewsListing } from '../../molecules/HackerNewsListing/HackerNewsListing'
import { GlobalContext } from '../../../store/context'
import Graph from '../../molecules/Graph/Graph'
import { Header } from '../../molecules/Header/Header'

import './hackerNews.scss'

export const HackerNews = () => {
  const { state } = useContext(GlobalContext)
  const { page } = state

  const [hits, sethits] = useState([])
  const [loadItems, setLoadItems] = useState(page)

  useEffect(() => {
    fetch(`${HACKER_NEWS}?page=${loadItems}`)
      .then(function (response) {
        if (response.status >= 400) {
          throw new Error('Bad response from server')
        }
        return response.json()
      })
      .then((data) => {
        sethits(data?.hits)
      })
  }, [loadItems])

  const hide = (id) => {
    const hideItem = hits.filter((itm) => itm.objectID !== id)
    sethits(hideItem)
  }

  const resultId = hits.map((obj) => {
    return {
      objectID: Object.values(obj.objectID).join(''),
      val: obj.points,
    }
  })

  return (
    <>
      <Header
        comments="Comments"
        voteCount="Vote count"
        upVote="Up vote"
        newsDetails="News Details"
      />
      <HackerNewsListing news={hits} hide={hide} />
      <div className="btn-container">
        <button
          type="button"
          disabled={loadItems === 1}
          onClick={() => setLoadItems(loadItems - 1)}
          className={`btn-container__btn ${
            loadItems === 1 ? 'disabled' : null
          }`}
        >
          Previous
        </button>
        <button
          type="button"
          onClick={() => setLoadItems(loadItems + 1)}
          className="btn-container__btn"
        >
          Next
        </button>
      </div>
      <Graph graphData={resultId} />
    </>
  )
}
