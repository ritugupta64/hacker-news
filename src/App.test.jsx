import React from 'react'
import { render } from '@testing-library/react'
import { BrowserRouter, Route, Switch } from 'react-router-dom'

import {GlobalContextProvider} from './store/context'
import {Header} from './components/molecules/Header/Header'
import {HackerRank} from './components/organism/HackerNews'

const dispatch = jest.fn()

const initialState = {
  character: [],
}

it('renders', () => {
  const { asFragment } = render(
    <GlobalContextProvider data={{ state: initialState, dispatch }}>
      <BrowserRouter>
        <Header
          comments="Comments"
          voteCount="Vote count"
          upVote="Up vote"
          newsDetails="News Details"
        />
        <Switch>
          <Route exact path="/" component={HackerRank} />
        </Switch>
      </BrowserRouter>
    </GlobalContextProvider>
  )
  expect(asFragment()).toMatchSnapshot()
})
