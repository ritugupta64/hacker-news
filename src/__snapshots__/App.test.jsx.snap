// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <main>
    <header
      class="header"
    >
      <div
        class="header__comments"
      >
        Comments
      </div>
      <div
        class="header__vote-count"
      >
        Vote count
      </div>
      <div
        class="header__up-vote"
      >
        Up vote
      </div>
      <div
        class="header__news-details"
      >
        News Details
      </div>
    </header>
    <div
      class="loader"
    />
    <div
      class="btn-container"
    >
      <button
        class="btn-container__btn disabled"
        disabled=""
        type="button"
      >
        Previous
      </button>
      <button
        class="btn-container__btn"
        type="button"
      >
        Next
      </button>
    </div>
    <div
      class="ct-chart "
    />
  </main>
</DocumentFragment>
`;
